application/src/test/java/com/sanction/lightning/resources/FacebookResourceTest.java

```diff
@@ -27,22 +29,26 @@
 
 public class FacebookResourceTest {
   private final ThunderClient thunderClient = mock(ThunderClient.class);
-  private final FacebookProviderFactory providerFactory = mock(FacebookProviderFactory.class);
-  private final FacebookProvider facebookProvider = mock(FacebookProvider.class);
-  private final UrlDownloadService urlDownloadService = mock(UrlDownloadService.class);
+  private final FacebookServiceFactory serviceFactory = mock(FacebookServiceFactory.class);
+  private final FacebookService facebookService = mock(FacebookService.class);
+  private final UrlService urlService = mock(UrlService.class);
   private final URLConnection urlConnection = mock(URLConnection.class);
+  private final InputStream inputStream = mock(InputStream.class);
+  private final FormDataContentDisposition contentDisposition =
+          mock(FormDataContentDisposition.class);
+  private final FacebookPhoto facebookPhoto = mock(FacebookPhoto.class);
```

Comment:
I think the `FacebookPhoto` object should be moved to the `publish()` method, since I think that is the only method that needs it.